### PR TITLE
[CI] Rename workflows/builds to tidy up check view

### PR DESCRIPTION
**Pull request by SaschaMann**
Some PRs like #2537 have a lot of checks. The check overview can get quite messy.

This PR renames all workflows/checks to follow the same scheme:

`category / name`

where `category` is currently `CI` (general checks like the file checks), `Tools` (Things like the summary generators, labeler) and track names.

The downside is that the detailed view and Action tab become less readable:

grafik
grafik

Personally I think the PR/Commit popup view are more important because they're the main thing one sees, so IMO this trade off is worth it.

---

Note that this will require changing the required checks to the new names (@iHiD)

### CODE REVIEW

File: .github/workflows/csharp-exercise-report.yml
Comment (iHiD) [inline .github/workflows/csharp-exercise-report.yml:14]:
```suggestion
    name: Update exercise report
```

If it's scoped to C#, do we need the extra "csharp" here?

File: .github/workflows/julia-exercise-ci.yml
Comment (iHiD) [inline .github/workflows/julia-exercise-ci.yml:15]: Depending on the result of the comment above, should "Julia" be added here.

File: .github/workflows/csharp-exercise-report.yml
Comment (SaschaMann) [reply .github/workflows/csharp-exercise-report.yml:14]: Nope, we don't, I just copied the old name. Good suggestion :)

File: .github/workflows/verify-csharp-formatting.yml
Comment (SaschaMann) [inline .github/workflows/verify-csharp-formatting.yml:13]:
```suggestion
    name: Verify formatting
```

**Review by iHiD — APPROVED** [review]
I think this is great. Thank you.

**Comment by SaschaMann** [comment]
The new required checks would be

- `CI / Check file sizes`
- `CI / Check for dead links`
- `CI / Verify formatting`
- `CI / Check file names`

I think

**Comment by SaschaMann** [comment]
This will also require a mass-rebase for open PRs so that the new required checks can run 🙃

**Review by ErikSchierboom — APPROVED** [review]
This is great!

**Comment by SaschaMann** [comment]
Thanks for the reviews :)

@iHiD you'll have to change the required checks before this can be merged

**Comment by iHiD** [comment]
@SaschaMann I think I've done it.

Screenshot 2020-10-27 at 16 52 57
Screenshot 2020-10-27 at 16 53 04
Screenshot 2020-10-27 at 16 53 09
Screenshot 2020-10-27 at 16 53 13
Screenshot 2020-10-27 at 16 53 17


Should I turn any of those off etc? Do I need to force-merge this? HELP! :)

**Comment by SaschaMann** [comment]
@iHiD Rebase, Format C# code, Format code and Page build are too much. Those are the /commands and github pages deploy

**Comment by SaschaMann** [comment]
Once the required checks in this PR all green, it works and it can be merged normally :)

**Comment by iHiD** [comment]
Sweet. Done. Thanks!

**Comment by SleeplessByte** [comment]
> This will also require a mass-rebase for open PRs so that the new required checks can run 🙃

Maybe this can be commented on all the PRs? Because I needlessly spent 30 minutes on this. Thanks

**Comment by SaschaMann** [comment]
> Maybe this can be commented on all the PRs? Because I needlessly spent 30 minutes on this. Thanks

I posted it to #v3 on Slack, but sure I'll do.

**Comment by SleeplessByte** [comment]
Thanks 👍 (the PR I pinged you on actually is a maintainer who isn't even on slack).

**Comment by SaschaMann** [comment]
I didn't think there were many maintainers that weren't on Slack, but I'll script a mass-comment tomorrow :)

**Pull request merged**
